test/unit/location_test.rb

```diff
@@ -28,108 +36,156 @@ def test_location_from_strange_hash
     assert_equal hash[:phone], location.phone
     assert_equal hash[:fax_number], location.fax
     assert_equal hash[:address_type].to_s, location.address_type
+    assert_equal hash[:name], location.name
   end
 
-  def test_pretty_print
-    expected = "110 Laurier Avenue West\nOttawa, ON, K1P 1J1\nCanada"
-    assert_equal expected, location_fixtures[:ottawa].prettyprint
+  test "#initialize sets the name to nil if it is not provided" do
+    location = Location.from({})
+    assert_nil location.name
   end
 
-  def test_to_s
-    expected = "110 Laurier Avenue West Ottawa, ON, K1P 1J1 Canada"
-    assert_equal expected, location_fixtures[:ottawa].to_s
+  test "#initialize sets company and company_name from company" do
+    location = Location.from(company: "Mine")
+
+    assert_equal "Mine", location.company
+    assert_equal "Mine", location.company_name
   end
 
-  def test_inspect
-    expected = "110 Laurier Avenue West\nOttawa, ON, K1P 1J1\nCanada\nPhone: 1-613-580-2400\nFax: 1-613-580-2495"
-    assert_equal expected, location_fixtures[:ottawa].inspect
+  test "#initialize sets company and company_name from company_name" do
+    location = Location.from(company_name: "Mine")
+
+    assert_equal "Mine", location.company
+    assert_equal "Mine", location.company_name
   end
 
-  def test_includes_name
-    location = Location.from(:name => "Bob Bobsen")
-    assert_equal "Bob Bobsen", location.name
+  test "#initialize prioritizes company" do
+    location = Location.from(company_name: "from company_name", company: "from company")
+
+    assert_equal "from company", location.company
+    assert_equal "from company", location.company_name
   end
 
-  def test_name_is_nil_if_not_provided
-    location = Location.from({})
-    assert_nil location.name
+  test "#prettyprint outputs a readable string" do
+    expected = "110 Laurier Avenue West\nOttawa, ON, K1P 1J1\nCanada"
+    assert_equal expected, @location.prettyprint
   end
 
-  def test_location_with_company_name
-    location = Location.from(:company => "Mine")
-    assert_equal "Mine", location.company_name
+  test "#to_s outputs a readable string without newlines" do
+    expected = "110 Laurier Avenue West Ottawa, ON, K1P 1J1 Canada"
+    assert_equal expected, @location.to_s
+  end
 
-    location = Location.from(:company_name => "Mine")
-    assert_equal "Mine", location.company_name
+  test "#inspect returns a readable string" do
+    expected = "110 Laurier Avenue West\nOttawa, ON, K1P 1J1\nCanada\nPhone: 1-613-580-2400\nFax: 1-613-580-2495"
+    assert_equal expected, @location.inspect
   end
 
-  def test_set_address_type
-    location = location_fixtures[:ottawa]
-    assert !location.commercial?
+  test "#address_type= assigns a type of address as commercial" do
+    refute @location.commercial?
```

Comment:
We should switch to `assert_predicate` here and everywhere else. Okay to leave this for a future PR, but best to create an issue if deferring. Feel free to assign to me.